website/content/en/blog/tracking-allocations.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Monitoring Vector's per-component memory usage
+short: Allocation tracking
+description: Gaining insight into Vector's per-component memory usage.
+authors: ["arshiyasolei", "tobz"]
+date: "2022-12-13"
+badges:
+  type: announcement
+  domains: ["monitoring"]
+tags: ["allocation-tracing", "tracking allocations"]
+---
+
+We are excited to announce that [Vector] now provides per-component memory usage metrics. To explore the feature, start a vector instance with the `allocation-tracing` option, launch `vector top` and monitor your components! This feature leverages a custom wrapping allocator combined with a separate reporting thread to collect our new metrics.
+
+## What's new
+
+We provide the following new metrics: `component_allocated_bytes`, `component_allocated_bytes_total`, and `component_deallocated_bytes_total`.
+
+- `component_allocated_bytes` captures the current net allocations/deallocations.
+- `component_allocated_bytes_total` shows the accumulated total allocations.
+- `component_deallocated_bytes_total` shows the accumulated total deallocations.
+
+## How it works
+
+From a high level point of view, we leverage our current component [tracing] infrastructure to track when a component "enters"/"exits" on a given thread. During each allocation/deallocation, we determine the responsible component via state stored in thread locals. Allocations that aren't associated with any components are tracked by a `root` component.
+
+## Notes
+
+This feature currently only supports unix based platforms. When enabled, there is approximately 20% overhead in throughput based on our benchmarks.
```

Comment:
```suggestion
## Using allocation tracing in production

Currently, this feature is only supported on Linux builds of Vector. This may change in the future as we continue to improve allocation tracing.

Additionally, there is a small performance overhead imposed by allocation tracing when enabled. Vector runs a set of soak tests in CI in order to catch performance regressions during development. These soak tests emulate common Vector use cases, such as remapping events, or converting payloads from one event type to another, and so on.

In our development and testing of this feature, we've observed around a 20% reduction in throughput. This should be low enough to allow enabling it on a single Vector instance for debugging purposes, but care should be taken before enabling it across your entire fleet.
```